### PR TITLE
basic repost ui

### PR DESCRIPTION
**Pull request by seanyesmunt**

### CODE REVIEW

File: ui/modal/modalRepost/view.jsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    title,
+    claim,
+    balance,
+    channels,
+    error,
+    reposting,
+    myClaims,
+    doFetchClaimListMine,
+  } = props;
+  const defaultName = claim && `${claim.name}-repost`;
```
Comment (kauffj) [inline ui/modal/modalRepost/view.jsx]: I think keep this as the same name, without `-repost`

File: ui/modal/modalRepost/view.jsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  const contentClaimId = claim && claim.claim_id;
+  const [repostChannel, setRepostChannel] = usePersistedState('repost-channel');
+  const [repostBid, setRepostBid] = usePersistedState('repost-bid', 0.1);
```
Comment (kauffj) [inline ui/modal/modalRepost/view.jsx]: 0.01

File: ui/modal/modalRepost/view.jsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    const channelToRepostTo = channels && channels.find(channel => channel.name === repostChannel);
+    if (channelToRepostTo && repostName && repostBid && repostChannel && contentClaimId) {
+      doRepost({
+        name: repostName,
+        bid: creditsToString(repostBid),
+        channel_id: channelToRepostTo.claim_id,
+        claim_id: contentClaimId,
+      }).then(() => {
+        doHideModal();
+        doToast({ message: __('Woohoo! Sucessfully reposted this claim.') });
```
Comment (kauffj) [inline ui/modal/modalRepost/view.jsx]: can link to your channel page

**Comment by seanyesmunt** [comment]
@kauffj Looking for better copy for https://github.com/lbryio/lbry-desktop/pull/3663/files#diff-c23b586b209fd42be1fa04108c651613R128 and https://github.com/lbryio/lbry-desktop/pull/3663/files#diff-c23b586b209fd42be1fa04108c651613R162

File: ui/modal/modalRepost/view.jsx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                  <FormField
+                    type="text"
+                    name="repost_name"
+                    value={repostName}
+                    error={repostNameError}
+                    onChange={event => setRepostName(event.target.value)}
+                  />
+                </fieldset-group>
+                <div className="form-field__help">
+                  {__('The name of your repost, something about reposting to help search')}
```
Comment (kauffj) [inline ui/modal/modalRepost/view.jsx]: Full text: Change this to repost to a different [community name](https://lbry.com/faq/naming).

File: ui/modal/modalRepost/view.jsx (hunk 3 above)
Comment (seanyesmunt) [reply ui/modal/modalRepost/view.jsx]: It won't be displayed on the channel page until it's confirmed

**Pull request merged**